imageio/imageio-pict/src/main/java/com/twelvemonkeys/imageio/plugins/pict/PICTImageReader.java

```diff
@@ -2058,10 +2057,10 @@ else if (packType == 4) {
         int[] pixArray = null;
         short[] shortArray = null;
         if (packType == 3) {
-            shortArray = new short[srcRect.height * (rowBytes + 1) / 2];
+            shortArray = new short[srcRect.height * (srcRect.width + 1)];
         }
         else {
-            pixArray = new int[srcRect.height * (rowBytes + 3) / 4];
+            pixArray = new int[srcRect.height * (srcRect.width + 3)];
```

Comment:
I don't think the +3 here is correct, unless you're also missing a / 4...? It was used in the original calculation to get the integer division truncation to "round" up.